components/copy_text.tsx

```diff
@@ -1,37 +1,31 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
-
 import {FormattedMessage} from 'react-intl';
 import {OverlayTrigger, Tooltip} from 'react-bootstrap';
 
 import Constants from 'utils/constants.jsx';
 import {copyToClipboard} from 'utils/utils.jsx';
 
-export default class CopyText extends React.PureComponent {
-    static propTypes = {
-        value: PropTypes.string.isRequired,
-
-        defaultMessage: PropTypes.string,
+type Props = {
+    value?: string;
```

Comment:
the value prop was "required" before a now is optional, why?

The `idMessage` was optional and now is required, why?

Reply:
> The idMessage was optional and now is required, why?

My ide gives the following warning.
```
TS2322: type 'string | undefined' is not assignable to type 'string' 
```

Reply:
Would it be a good idea to remove `defaultProps` and make the following changes to resolve this?

```typescript
type Props = {
     value: string;
     defaultMessage ?: string | 'Copy';
     idMessage: string | 'integrations.copy';
};
```

Reply:
If you are using defaultProps I think you no longer need any of them to be option, so you can remove the "optional" part of the defaultMessage too.

Reply:
Bottom line does it mean to delete optional from `defaultMessage`, `idMessage`?

Reply:
Nope, I think for our use case both should be mandatory

Reply:
Sure, they weren't mandatory before, but actually `idMessage` is mandatory for `FormattedMessage`, so it was kinda mandatory, so you did well making it mandatory. But `defaultMessage` is optional for `FormattedMessage`, but we always provide the id and the default message, so I think is better to set the `defaultMessage` as mandatory here.

Reply:
Thanks you for detail explain! :)